drivers/dac/ad5754r/ad5754r.c

```diff
@@ -0,0 +1,684 @@
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_set_sdo_disable(struct ad5754r_dev *dev,
+			    enum ad5754r_ctrl_sdo_dis sdo_dis)
+{
+	int ret;
+
+	if (!dev || (sdo_dis > AD5754R_CTRL_SDO_DIS))
+		return -EINVAL;
+
+	ret = ad5754r_update_bits(dev,
+				  AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_CONTROL,
+						  AD5754R_TSD_CLAMP_CLR_SDO_DAC_ADDR),
+				  AD5754R_CTRL_SDO_DISABLE_MASK,
+				  sdo_dis);
+	if (ret)
+		return ret;
+
+	dev->sdo_dis = sdo_dis;
+
+	return 0;
+}
+
+/**
+ * @brief Set Internal Reference Enable/Disable state for device.
+ * @param dev - The device structure.
+ * @param int_ref_pwrup - Internal Reference Enable/Disable state for device.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_set_int_ref_pwrup(struct ad5754r_dev *dev,
+			      enum ad5754r_pwr_int_ref_pwrup int_ref_pwrup)
+{
+	int ret;
+
+	if (!dev || (int_ref_pwrup > AD5754R_PWR_INT_REF_POWERUP))
+		return -EINVAL;
+
+	// TODO Confirm with apps on whether the DAC channel address
+	// is don't care, or 0
+
+	ret = ad5754r_update_bits(dev,
+				  AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_PWR_CTRL, 0),
+				  AD5754R_PWR_UP_INT_REF_MASK,
+				  int_ref_pwrup);
+	if (ret)
+		return ret;
+
+	dev->int_ref_pwrup = int_ref_pwrup;
+
+	return 0;
+}
+
+/**
+ * @brief Get Over-current status bit for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param oc_ch_alert - Store channel overcurrent alert status.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_get_oc_ch_alert(struct ad5754r_dev *dev, uint8_t chn,
+			    enum ad5754r_pwr_oc_ch_alert *oc_ch_alert)
+{
+	if (!dev || !oc_ch_alert || (chn >= AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	// TODO Confirm with apps on whether the DAC channel address
+	// is don't care, or 0
+
+	return ad5754r_read_masked(dev,
+				   AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_PWR_CTRL, 0),
+				   AD5754R_PWR_OC_ALERT_CH_MASK(chn),
+				   (uint16_t *)oc_ch_alert);
+}
+
+/**
+ * @brief Get Thermal Shutdown status bit.
+ * @param dev - The device structure.
+ * @param tsd_alert - Store TSD alert status.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_get_tsd_alert(struct ad5754r_dev *dev,
+			  enum ad5754r_pwr_tsd_alert *tsd_alert)
+{
+	if (!dev || !tsd_alert)
+		return -EINVAL;
+
+	// TODO Confirm with apps on whether the DAC channel address
+	// is don't care, or 0
+
+	return ad5754r_read_masked(dev,
+				   AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_PWR_CTRL, 0),
+				   AD5754R_PWR_TSD_ALERT_MASK,
+				   (uint16_t *)tsd_alert);
+}
+
+/**
+ * @brief Write to and read back a register to verify SPI.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_spi_verify(struct ad5754r_dev *dev)
+{
+	int ret;
+	uint16_t reg_data = 0;
```

Comment:
is this explicit initialization required?

Reply:
Nope, have removed the explicit initialization